domain/machine/state/types.go

```diff
@@ -48,15 +63,15 @@ func tagsFromHardwareCharacteristics(machineUUID string, hc *instance.HardwareCh
 	return res
 }
 
-func (d *instanceData) toHardwareCharacteristics() *instance.HardwareCharacteristics {
+func (d *instanceDataResult) toHardwareCharacteristics() *instance.HardwareCharacteristics {
```

Comment:
One for clean up. The core packages should not be used in state. It's ok in services, but not in state. We should serialize from machine types to state types. That way we "own" the types, so we have the liberty to migrate the database without changing the types, or vice versa.